### PR TITLE
Sandy phong submit surveys and modeling

### PR DESCRIPTION
**Pull request by ptt5039**

### CODE REVIEW

File: src/main/java/com/revature/controllers/HistoryController.java [hunk 1]
```diff
@@ -46,7 +44,8 @@ public History findById(@PathVariable int id) {
 	
 	// Query for email. Uses post mapping so that the @ symbol can be send in the body rather than in the URL
 	@PostMapping("/email")
-	public List<History> findByEmail(@Valid @RequestBody String email) {
+	public List<History> findByEmail(@RequestBody String email) {
+		System.out.println(email);
```
Comment (btkruppa) [inline src/main/java/com/revature/controllers/HistoryController.java]: remove sysout

File: src/main/java/com/revature/controllers/SurveyController.java [hunk 2]
```diff
@@ -29,6 +29,16 @@
 	public List<Survey> findAll() {
 		return surveyService.findAllOrderByDateCreatedDesc();
 	}
+	
+	@GetMapping("template")
```
Comment (btkruppa) [inline src/main/java/com/revature/controllers/SurveyController.java]: should be /surveys/template/{isTemplate}
where true or false is passed in

File: src/main/java/com/revature/controllers/SurveyController.java [hunk 3]
```diff
@@ -29,6 +29,16 @@
...
+	public List<Survey> findAllTemplate() {
+		return surveyService.findAllTemplateOrderByDateCreatedDesc();
+	}
+	
+	@GetMapping("published")
```
Comment (btkruppa) [inline src/main/java/com/revature/controllers/SurveyController.java]: should be done similar to the template endpoint

File: src/main/java/com/revature/controllers/HistoryController.java (hunk 1 above)
Comment (btkruppa) [inline src/main/java/com/revature/controllers/HistoryController.java]: remove sysout

File: src/main/java/com/revature/repos/SurveyRepo.java [hunk 4]
```diff
@@ -17,5 +17,11 @@
 
 	@Query("From Survey ORDER BY dateCreated DESC")
 	List<Survey> findAllOrderByDateCreatedDes();
+	
+	@Query("FROM Survey s WHERE s.template = true ORDER BY dateCreated DESC")
```
Comment (btkruppa) [inline src/main/java/com/revature/repos/SurveyRepo.java]: try not to use query for this, make name findByTemplateOrderByDateCreatedDesc(boolean template) and it should work, then in your service layer you can pass in true or false, also might want to add pagination so perhaps Page< List < Survey > > findByTemplateOrderByDateCreatedDesc(boolean template, Pageable page)

File: src/main/java/com/revature/repos/SurveyRepo.java [hunk 5]
```diff
@@ -17,5 +17,11 @@
...
+	List<Survey> findAllTemplateOrderByDateCreatedDes();
+	
+	@Query("FROM Survey s WHERE s.published = true ORDER BY dateCreated DESC")
```
Comment (btkruppa) [inline src/main/java/com/revature/repos/SurveyRepo.java]: similar feedback to template one

File: src/main/java/com/revature/services/SurveyServiceImpl.java [hunk 6]
```diff
@@ -49,15 +64,51 @@ public Survey findById(int id) {
...
+			newSqj.setSurvey(newSurvey);
+			newSqj.setQuestionOrder(i + 1);
+		}
+
+		Survey survey = surveyRepo.getOne(newSurvey.getSurveyId());
+
+		return survey;
 	}
 
+//	@Override
```
Comment (btkruppa) [inline src/main/java/com/revature/services/SurveyServiceImpl.java]: delete unused commented out code

**Review by btkruppa — CHANGES_REQUESTED** [review]
looks pretty good but try fixing the few things I added comments to

File: src/main/java/com/revature/controllers/SurveyController.java (hunk 2 above)
Comment (ptt5039) [reply src/main/java/com/revature/controllers/SurveyController.java]: done

File: src/main/java/com/revature/controllers/SurveyController.java (hunk 3 above)
Comment (ptt5039) [reply src/main/java/com/revature/controllers/SurveyController.java]: removed published endpoint

File: src/main/java/com/revature/repos/SurveyRepo.java (hunk 4 above)
Comment (ptt5039) [reply src/main/java/com/revature/repos/SurveyRepo.java]: done

**Pull request merged**